modules/code-editor/src/backend/editor.ts

```diff
@@ -58,23 +59,67 @@ export default class Editor {
       throw new Error('Invalid hook type.')
     }
 
-    if (!FILENAME_REGEX.test(name)) {
-      throw new Error('Filename has invalid characters')
+    this._validateFilename(name)
+  }
+
+  private _validateFilename(filename: string) {
+    if (!FILENAME_REGEX.test(filename)) {
+      throw new EditorError('Filename has invalid characters', EditorErrorStatus.INVALID_NAME)
     }
   }
 
+  private _loadGhostForBotId(file: EditableFile): sdk.ScopedGhostService {
```

Comment:
Not very important, but the naming isn't very clear on what that method does, where is the bot id? It would be clearer by calling this._loadGhostForBotId(file.botId) instead and using private _loadGhostForBotId(botId: string | undefined).